pyomo/network/util.py

```diff
@@ -25,24 +25,25 @@ def replicate_var(comp, name, block, index_set=None):
             index_set = UnindexedComponent_set
 
     var_args = {}
-    try:
-        var_args['domain'] = comp.domain
-    except AttributeError:
-        pass
-    try:
-        var_args['bounds'] = comp.bounds
-    except AttributeError:
-        pass
+    # try:
+    #     var_args['domain'] = comp.domain
+    # except AttributeError:
+    #     pass
+    # try:
+    #     var_args['bounds'] = comp.bounds
+    # except AttributeError:
+    #     pass
```

Comment:
So this part is related to #893, and @jsiirola I think it'd be good for you to weigh in.

I can definitely see the problem with copying bounds to expanded variables and perhaps even implicit/auto variables, especially with the introduction of potentially negative variables. However, I think there is also a problem if we chose to skip copying the domain to the new variables, since I can imagine we don't want to be equating reals with integers and then passing that to our solvers, if we can help it. Copying domains should present less of a problem than bounds since they can be as inclusive as Reals or Integers. Although, you could also have domains with inherent bounds like NonNegativeReals or Binary. I would think if the bounds inherent in those domains starting causing problems for a user, it would be the user's responsibility to change the domain of the parent variable to be more inclusive. Does that make sense?

If we agree to keep copying domains but remove copying bounds, then I would say uncomment the domain parts, delete the now commented-out bounds parts, and update any comments/docstring in this function that refers to copying bounds.

I also took a look at the failing tests, they all have to do with checking to make sure domains and bounds were copied, so if we go with the above plan then you should go into the failing tests in `test_arc.py` and remove the assertions for bounds.

Reply:
Copying domain is a good compromise indeed, and it make sense. 
Thanks for looking to the failing test. This PR should be fine after changing it.

Reply:
Setting the domain can also be problematic.  Consider the case of an integer Port with two Arcs leaving it.  There is no reason to believe that the arc flow rates should be restricted to discrete values - only that the sum be integer.

This does beg the question, though if we need to augment the API to allow the user to specify domains / bounds for arc variables?

Reply:
It's technically possible already (and not altogether too difficult) to modify the domains and such of arc variables. All the user needs to do is `m.myarc.flow.domain = NonNegativeReals` (or whatever the right syntax is) because Arc's getattr method:

> Returns `self.expanded_block.name` if it exists